Source/SelfService/Web/application/create.tsx

```diff
@@ -317,64 +198,183 @@ export const Create: React.FunctionComponent<Props> = (props) => {
             };
             try {
                 await createApplication(input);
-                // redirect to build
                 const href = `/application/building/${application.id}`;
                 history.push(href);
-
+                setServerError(false);
                 enqueueSnackbar('Application created', { variant: 'info' });
             } catch (error) {
-                enqueueSnackbar('Error creating application', { variant: 'error' });
+                setServerError(true);
             }
+
             return;
         }
 
-        setActiveStep((prevActiveStep) => prevActiveStep + 1);
+        return;
+    };
+
+    const validateForm = () => {
```

Comment:
This file is getting big. Do you think we should move this validation function into its own file and then import it here? Perhaps create a folder called helperFunctions and put it inside there? This will require rewriting it a bit to pass all the necessary arguments to it when it is being called. Thoughts?

Reply:
Really good idea :)